### PR TITLE
Refactor collections

### PR DESCRIPTION
**Pull request by aperezsantos**
Moved methods and tests from stat tracker to their respective collections

### CODE REVIEW

File: lib/game_collection.rb [hunk 1]
```diff
@@ -14,61 +14,61 @@ def create_games(game_data)
...
   def lowest_total_score
-    games.map do |game|
+    @games.map do |game|
       game.home_goals + game.away_goals
     end.min
   end
 
   def biggest_blowout
-    games.map do |game|
+    @games.map do |game|
```
Comment (colintalex) [inline lib/game_collection.rb:29]: Glad we changed all these!

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -201,30 +173,15 @@ def team_name_by_id(team_id)
...
-    games = find_games_in_season(season)
-    totals = Hash.new(0)
-    games.each do |game|
-      totals[game.team_id] += game.tackles
-    end
-    team_name_by_id(totals.key(totals.values.min))
+    team_name_by_id(gtc.season_tackles(season).key(gtc.season_tackles(season).values.min))
   end
 
 ###### Helpful IT4 methods
```
Comment (mikez321) [inline lib/stat_tracker.rb:183]: I think we're almost to a point that we can get rid of these comments

**Review by mikez321 — APPROVED** [review]

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -76,50 +76,28 @@ def count_of_teams
...
-      game_team.team_id
-    end
-    games_per_team = game_teams_grouped_by_team_id.each_pair do |team_id, games_by_team|
-      total_goals = games_by_team.map do |single_game|
-        single_game.goals
-      end
-      game_team_averages = (total_goals.sum.to_f / total_goals.length).round(2)
-      (game_teams_grouped_by_team_id[team_id] = game_team_averages)
-    end
-    team_name_by_id(games_per_team.key(games_per_team.values.max))
```
Comment (colintalex) [inline lib/stat_tracker.rb:89]: So pretty and small!

File: test/stat_tracker_test.rb [hunk 4]
```diff
@@ -104,7 +104,7 @@ def test_it_knows_the_hightest_scoring_home_team
   #harness pass
 end
 
-def test_it_can_return_the_best_overall_defense
```
Comment (colintalex) [inline test/stat_tracker_test.rb:107]: Good rename

**Review by colintalex — APPROVED** [review]

**Pull request merged**